debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/RecordMakers.java

```diff
@@ -53,6 +53,7 @@
     private final Map<CollectionId, RecordsForCollection> recordMakerByCollectionId = new HashMap<>();
     private final Function<Document, String> valueTransformer;
     private final BlockingConsumer<SourceRecord> recorder;
+    private final boolean compactOnDelete;
```

Comment:
I'd call it `(emit)TombstonesOnDelete`, because that's what we can control. The aspect of compaction is a consequence of it left to Kafka.